python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -82,6 +82,9 @@ def __init__(self, data_context: DataContext, dataset_tag: str = "unknown_datase
         self._num_errored_blocks = 0
 
         self._last_debug_log_time = 0
+        # Exception that caused the executor to fail.
+        # None if the executor succeeded or is still running.
+        self._exception: Optional[Exception] = None
```

Comment:
Where do we use the `_exception` attribute? I think we might just set it but not actually use it anywhere?

Reply:
let me remove it. initially I thought it'd be used.